src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -207,6 +207,33 @@ public void Deserialize(byte[] adalState)
             }
         }
 
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <returns></returns>
+        public byte[] SerializeV3()
+        {
+            lock (cacheLock)
+            {
+                var jsonSerializer = new TokenCacheJsonSerializer(tokenCacheAccessor);
+                return jsonSerializer.Serialize();
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="bytes"></param>
+        public void DeserializeV3(byte[] bytes)
```

Comment:
this needs an additional param, called merge (default should potentially be false as it is currently in MSAL v3)

Reply:
as discussed, we're not doing this.

---
In reply to: [260968891](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1500#discussion_r260968891) [](ancestors = 260968891)